src/lib/common/analyser.py

```diff
@@ -194,25 +199,32 @@ def __get_all_media(self):
         return all_media
 
     def _run(self, config):
-        self.setup_run()
+        self.pre_analyse(config)
 
         all_media = self.__get_all_media()
         elements = self.__get_elements(all_media)
 
+        derived_folders = set([])
         for element in elements:
             # TODO: create try/catch infrastructure to delete this folder if there is an error.
             if not os.path.exists(element["dest"]):
                 os.makedirs(element["dest"])
 
             self.analyse_element(element, config)
+            derived_folders.add(element["derived_folder"])
 
+        self.post_analyse(config, derived_folders)
         save_logs(self.__logs, self.ANALYSER_LOGS)
 
-    def setup_run(self):
+    def pre_analyse(self, config):
```

Comment:
Yes, this is a great idea. Could you update the other analyser modules that used `setup_run` to use `pre_analyse` instead (adding the `config` arg as well)?

Reply:
Thought I had done this - managed to miss the OCR analyser. Updated.